src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala

```diff
@@ -153,7 +157,56 @@ class WorkflowController @Inject()(workflowService: WorkflowService) {
   @PostMapping(path = Array("/workflow/import"))
   def importWorkflow(@RequestPart("file") file: MultipartFile): CompletableFuture[WorkflowJoined] = {
     val workflowImport = ObjectMapperSingleton.getObjectMapper.readValue(file.getBytes, classOf[WorkflowImportExportWrapper])
-    workflowService.importWorkflow(workflowImport).toJava.toCompletableFuture
+    workflowService.convertToWorkflowJoined(workflowImport).toJava.toCompletableFuture
+  }
+
+  @PostMapping(path = Array("/workflows/import"))
+  def importWorkflows(@RequestPart("file") file: MultipartFile): CompletableFuture[Seq[Project]] = {
+    val zipEntries = extractZipEntries(file.getBytes)
+    val workflowImports = zipEntries.map {
+      case (entryName, byteArray) => entryName -> Try(
+        ObjectMapperSingleton.getObjectMapper.readValue(byteArray, classOf[WorkflowImportExportWrapper])
+      )
+    }
+
+    val parsingErrors = workflowImports
+      .filter { case (_, readTry) => readTry.isFailure }
+      .map { case (entryName, readTry) =>
+        logger.error(s"Could not parse $entryName", readTry.failed.get)
+        BulkOperationError(entryName, GenericError("An error occurred while parsing this entry."))
+      }
+    if (parsingErrors.nonEmpty) {
+      throw new ApiException(parsingErrors)
+    }
+
+    workflowService.importWorkflows(workflowImports.map(_._2.get)).toJava.toCompletableFuture
+  }
+
+  private def extractZipEntries(zipByteArray: Array[Byte]) = {
+    val bais = new ByteArrayInputStream(zipByteArray)
+    val zis = new ZipInputStream(bais)
+    val zipEntries = ArrayBuffer[(String, Array[Byte])]()
+    var entry = zis.getNextEntry
+    while (entry != null) {
+      if (!entry.getName.startsWith("__MACOSX/")) {
```

Comment:
Could you please explain this line ?

Reply:
If you create a zip file manually on mac, it creates a hidden folder called __MACOSX. The content of that folder should be ignored.

Reply:
Cool